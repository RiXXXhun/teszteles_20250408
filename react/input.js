import { useState } from "react"

export default function input() {
    const [input, setInput] = useState("");
    const [kiir, setKiir] = useState("");
    return (
        <div>
            <input type="text" data-testid="input" value={input} onChange={(event) => {setInput(event.target.value)}} />
            <button data-testid="button" onClick={() => {
                    setKiir(input)
                }}
            >
                KiÃ­r
            </button>
            <p data-testid="p">
                {kiir.length > 0 && `${kiir}`}
            </p>
        </div>

    )
}