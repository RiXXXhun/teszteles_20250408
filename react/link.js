import { useState } from "react"

const statuses = {
    HOVER: "hovered",
    NORMAL: ""
}

export default function Link({
    page,
    children
}) {
    const [status, setStatus] = useState();

    return (
        <a
            data-testid="link"
            href={page || "#"}
            className="status"
            onMouseEnter={() => {
                setStatus(statuses.HOVER)
            }}
            onMouseLeave={() => {
                statuses(statuses.NORMAL)
            }}
        >
            {children}
        </a>
    )
}