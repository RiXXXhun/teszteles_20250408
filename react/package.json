{
  "name": "react",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "test": "jest --collect-coverage"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/user-event": "^14.6.1",
    "jest-environment-jsdom": "^29.7.0",
    "react": "^19.1.0",
    "react-dom": "^19.1.0"
  },
  "jest": {
    "testEnvironment": "jsdom"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.26.9",
    "@babel/preset-react": "^7.26.3",
    "@testing-library/dom": "^10.4.0",
    "@testing-library/react": "^16.3.0",
    "babel-jest": "^29.7.0",
    "jest": "^29.7.0",
    "react-test-renderer": "^19.1.0"
  }
}
