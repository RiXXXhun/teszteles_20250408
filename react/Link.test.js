import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'
import Link from './link'


it('changes class on hover', async () => {
    render(<Link page="https://google.com">Google</Link>)

    expect(screen.getByTestId("link")).toHaveTextContent("Google")

    await userEvent.hover(screen.getByTestId('link'))

    expect(screen.getByTestId("link").className).toBe("hovered")
})