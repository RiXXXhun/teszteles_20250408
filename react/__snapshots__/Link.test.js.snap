// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes class on hover 1`] = `null`;
